### PR TITLE
Add collaborators to TodoistClient

### PR DESCRIPTION
**Pull request by AhmedZaki99**
`TodoistClient` is missing two sharing-related resources: [`Collaborators`](https://developer.todoist.com/sync/v9/#collaborators) and [`Collaborator States`](https://developer.todoist.com/sync/v9/#collaborator-states). So, the following changes were made:

First, models are added for both resources and included in the `Resources` and `ResourceType` classes.

Second, the section resource type was missing from the `ResourceType` enum, so it has been added.

Last, a `CollaboratorsService` has been added to `TodoistClient` along with its tests.

### CODE REVIEW

File: src/Todoist.Net.Tests/Services/CollaboratorsServiceTests.cs
Comment (olsh) [inline src/Todoist.Net.Tests/Services/CollaboratorsServiceTests.cs:1]: Let's use `async`/`await` for the new code.

File: src/Todoist.Net/Models/CollaboratorState.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        /// <value>The state.</value>
+        [JsonProperty("state")]
+        public string State { get; internal set; }
+
+        /// <summary>
+        /// Gets a value indicating whether this collaborator is deleted.
+        /// </summary>
+        /// <value><c>true</c> if this collaborator is deleted; otherwise, <c>false</c>.</value>
+        [JsonProperty("is_deleted")]
+        public string IsDeleted { get; internal set; }
```
Comment (olsh) [inline src/Todoist.Net/Models/CollaboratorState.cs:40]: This should be boolean

File: src/Todoist.Net/Models/CollaboratorState.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        /// <value>The project id.</value>
+        [JsonProperty("project_id")]
+        public string ProjectId { get; internal set; }
+
+        /// <summary>
+        /// Gets the collaborator state.
+        /// </summary>
+        /// <value>The state.</value>
+        [JsonProperty("state")]
+        public string State { get; internal set; }
```
Comment (olsh) [inline src/Todoist.Net/Models/CollaboratorState.cs:33]: Let's create a string enum for this

**Review by olsh — CHANGES_REQUESTED** [review]
I'd suggest to rename the service name to `Sharing` and implement the other methods described under the `Sharing` section.
https://developer.todoist.com/sync/v9/#collaborators

**Comment by olsh** [comment]
Thank you for the contribution.

File: src/Todoist.Net.Tests/Services/CollaboratorsServiceTests.cs
Comment (AhmedZaki99) [reply src/Todoist.Net.Tests/Services/CollaboratorsServiceTests.cs:1]: Done 👍

File: src/Todoist.Net/Models/CollaboratorState.cs (hunk 1 above)
Comment (AhmedZaki99) [reply src/Todoist.Net/Models/CollaboratorState.cs]: My bad,
I fixed it now.

File: src/Todoist.Net/Models/CollaboratorState.cs (hunk 2 above)
Comment (AhmedZaki99) [reply src/Todoist.Net/Models/CollaboratorState.cs]: Done 👍
I created a new StringEnum class called `CollaboratorStatus`

**Comment by AhmedZaki99** [comment]
> I'd suggest to rename the service name to `Sharing` and implement the other methods described under the `Sharing` section. https://developer.todoist.com/sync/v9/#collaborators

Ok, I am working on it. @olsh

**Comment by AhmedZaki99** [comment]
@olsh 

Alright, here's what I found:
- I found a [`SharingService`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Services/SharingService.cs) already added to the project that extends the [`SharingCommandService`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Services/SharingCommandService.cs) but doesn't have any extra functionalities (no methods not being sent by sync transactions).
- The interface it implements, [`ISharingService`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Services/ISharingService.cs), for some reason didn't inherit the [`ISharingCommandService`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Services/ISharingCommandService.cs) interface, but instead, had all methods repeated.

So, what I did was the following:
- Moving the methods I created in the `CollaboratorsService` to the `SharingService` and removing repeated methods in `ISharingServiec` by inheritance.
- Deleting the `CollaboratorsService` along with its interface and tests.
- Extending the existing test in the `SharingServiceTests` class to check collaborator methods.
- Changing the test to use async\await and try\finally blocks to ensure that loose ends are cleaned up properly.

But it was great that you noticed all `Sharing` methods should belong to the same service, or if we added the `CollaboratorsService` as it is, moving methods back to the `SharingService` would've been a breaking change.

**Pull request merged**